mycroft/util/log.py

```diff
@@ -51,99 +50,49 @@ def method(cls, *args, **kwargs):
-    """
-
-    _custom_name = None
-    handler = None
-    level = None
-
-    # Copy actual logging methods from logging.Logger
-    # Usage: LOG.debug(message)
-    debug = _make_log_method(logging.Logger.debug)
-    info = _make_log_method(logging.Logger.info)
-    warning = _make_log_method(logging.Logger.warning)
-    error = _make_log_method(logging.Logger.error)
-    exception = _make_log_method(logging.Logger.exception)
-
-    @classmethod
-    def init(cls):
-        """ Initializes the class, sets the default log level and creates
-        the required handlers.
-        """
-
-        # Check configs manually, the Mycroft configuration system can't be
-        # used since it uses the LOG system and would cause horrible cyclic
-        # dependencies.
-        confs = [SYSTEM_CONFIG, USER_CONFIG]
-        config = {}
-        for conf in confs:
-            try:
-                merge_dict(config,
-                           load_commented_json(conf) if isfile(conf) else {})
-            except Exception as e:
-                print('couldn\'t load {}: {}'.format(conf, str(e)))
-
-        cls.level = logging.getLevelName(config.get('log_level', 'INFO'))
-        log_message_format = (
-            '{asctime} | {levelname:8} | {process:5} | {name} | {message}'
-        )
-
-        formatter = logging.Formatter(log_message_format, style='{')
-        formatter.default_msec_format = '%s.%03d'
-        cls.handler = logging.StreamHandler(sys.stdout)
-        cls.handler.setFormatter(formatter)
-
-        # Enable logging in external modules
-        cls.create_logger('').setLevel(cls.level)
-
-    @classmethod
-    def create_logger(cls, name):
-        logger = logging.getLogger(name)
-        logger.propagate = False
-        logger.addHandler(cls.handler)
-        return logger
-
-    def __init__(self, name):
-        LOG._custom_name = name
-
-    @classmethod
-    def _log(cls, func, *args, **kwargs):
-        if cls._custom_name is not None:
-            name = cls._custom_name
-            cls._custom_name = None
-        else:
-            # Stack:
-            # [0] - _log()
-            # [1] - debug(), info(), warning(), or error()
-            # [2] - caller
-            try:
-                stack = inspect.stack()
-
-                # Record:
-                # [0] - frame object
-                # [1] - filename
-                # [2] - line number
-                # [3] - function
-                # ...
-                record = stack[2]
-                mod = inspect.getmodule(record[0])
-                module_name = mod.__name__ if mod else ''
-                name = module_name + ':' + record[3] + ':' + str(record[2])
-            except Exception:
-                # The location couldn't be determined
-                name = 'Mycroft'
-
-        func(cls.create_logger(name), *args, **kwargs)
-
-
-LOG.init()
+LOG = logging.getLogger('Mycroft')
+
+
+def create_logger(name):
```

Comment:
If we are going to do this, I would like to see a daily rotating file handler added (there is one built into python).  the current log files can get ungainly big.  this would mean removing shell script code that redirects STDOUT to the log file.  Another benefit to this is that the "log" file would only have log messages in it.  If we still want to capture the print statements and such, that should be written to a different file.  a quick cron job to delete older daily log files would be a good idea too.  it would take a while, but losing a device due to log file disk consumption would be bad.

Reply:
No that's a system level implementation. service or running script that should handle. logging should be trivial so it can be redirected and modified depending on system implementation.

Reply:
I am not sure I understand the delineation you are trying to make here.  Skills, voice, audio, enclosure... these are all services and we have every right to control how the logging occurs for those services as we are the custodians of them. 

If we are going to allow skill developers to produce logs based on their personal preferences,  which seems to be what you are going for based on your comments, skill logs should be independent of core logs.  Each skill should have its own logger instance with messages that are not interleaved with other logs emitted by core.